src/ir/proto/sql/decode.cc

```diff
@@ -0,0 +1,42 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#include "src/ir/proto/sql/decode.h"
+
+namespace raksha::ir::proto::sql {
+
+ir::Value DecodeSourceTableColumn(
+    const SourceTableColumn &source_table_column,
+    IRContext &ir_context) {
+  const std::string &column_path = source_table_column.column_path();
+  CHECK(!column_path.empty()) << "Required field column_path was empty.";
+  // Note: In the future, we will probably want to make a `Storage` per table
+  // and grab columns from that table using operators. For now, however,
+  // making each column an independent storage works for the MVP and is much
+  // simpler.
+  //
+  // Also, for now, we consider all storages to have primitive type. We will
+  // probably want to change that when we start handling types in a
+  // non-trivial fashion.
+  return ir::Value{value::StoredValue(
+      ir_context.GetOrCreateStorage(
```

Comment:
I commented on the parent PR. Ideally, we should setup the `IRContext` with all the `Operators` and `Storage` and use it here. Adding things on the fly can be error prone. e.g., if we add more attributes to `Storage`, how do we ensure that all uses are consistent. 

With `GetOrCreateXYZ` variants, it looks like the first occurrence wherever it is wins?

Reply:
Yes. One way to guard against the errors that you are describing might be to add a `isStaticallyIdentical` option to `Value`s, which indicates if the item is known at static time to be an identical value. Then, if we attempt to call `GetOrCreateNode` with a candidate `Node` and that `Node` is not `isStaticallyIdentical` to the value currently in the map, we can error.

For `Operator` and `Storage`, `isStaticallyIdentical` would currently just be that the names (and in the case of `Storage`, types) match. We can default it to `false` for everything else until we need this concept there.

Reply:
This is now fixed.